pkg/github/structs.go

```diff
@@ -15,9 +40,196 @@ type GhCheckRun struct {
 }
 
 type CheckEvent struct {
-	Repo       Repo
-	Sha        string
-	Token      string
-	HeadBranch string
-	GhClient   *github.Client
+	GenericGithubEvent
+	repo     Repo
+	sha      string
+	token    string
+	branch   string
+	prUrl    string
+	ghClient *github.Client
```

Comment:
we could probably include a context in this struct and pass it to the rest of the checks

Reply:
why not, let's discuss it IRL